src/python/pants/backend/python/subsystems/pex_build_util.py

```diff
@@ -55,15 +55,27 @@ def has_python_requirements(tgt):
   return isinstance(tgt, PythonRequirementLibrary)
 
 
-def can_have_python_platform(tgt):
-  return isinstance(tgt, (PythonBinary, PythonDistribution))
+def may_add_python_platform_to_resolve(tgt):
+  return isinstance(tgt, (PythonBinary, PythonDistribution, PythonTests))
 
 
 def targets_by_platform(targets, python_setup):
   d = defaultdict(OrderedSet)
   for target in targets:
-    if can_have_python_platform(target):
-      for platform in target.platforms if target.platforms else python_setup.platforms:
+    if may_add_python_platform_to_resolve(target):
+      # PythonTests doesn't have a platforms property, and should (currently) always run on the
```

Comment:
Is this better than saying explicitly that `PythonTests` should always fall back to default platforms? Something like:
```
if has_to_run_under_default_platform(target): # PythonTests case
  for platform in python_setup.platforms:
    # do something
elif may_add_python_platform_to_resolve(target):
  # Do something

```

Then, this comment could move to the docstring of has_to_run_under_default_platform, and we avoid some possibly stale code.

That said, this is probably bike-sheddy :)

Reply:
Yeah, I was 50:50 on this... I prefer splitting out the separate case, as it means I can delete a lot of comments. Done.